source/uwp/Renderer/lib/XamlBuilder.h

```diff
@@ -47,11 +47,11 @@ namespace AdaptiveNamespace
                                    _In_ ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
                                    _In_ ABI::AdaptiveNamespace::IAdaptiveRenderArgs* renderArgs,
                                    _Outptr_ ABI::Windows::UI::Xaml::IUIElement** textBlockControl);
-        static void BuildContainer(_In_ ABI::AdaptiveNamespace::IAdaptiveCardElement* adaptiveCardElement,
```

Comment:
>static [](start = 8, length = 6)

I'd prefer we keep these static if we can. What did you need that required making this non-static?

Reply:
`XamlBuilder::BuildContainer()` and `XamlBuilder::BuildColumn()` now call `XamlBuilder::ApplyBackgroundToRoot()` (non-static).

I noticed a somewhat related issue, though. `XamlBuilder::BuildPanelChildren()` calls `AdaptiveContainerRenderer::Render()` which just calls `XamlBuilder::BuildContainer()`. Originally, I created a new instance of `XamlBuilder` to be able to do this. I'm realizing that instead I should be using the main/only instance of it.

Reply:
My ActionSet change (now in master) updated ApplyBackgroundToRoot to be static. Try merging that in and see if you can keep BuildContainer and BuildColumn static.

---
In reply to: [249991803](https://github.com/Microsoft/AdaptiveCards/pull/2316#discussion_r249991803) [](ancestors = 249991803)